### PR TITLE
Add with-markdoc/ example

### PR DESCRIPTION
**Pull request by mfix-stripe**
## Documentation / Examples

- [x] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: examples/with-markdoc/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start"
+  },
+  "dependencies": {
+    "@markdoc/markdoc": "*",
```
Comment (leerob) [inline examples/with-markdoc/package.json:9]: Should we just do latest instead of star?

File: examples/with-markdoc/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Next.js example
+
+This is a boilerplate for a creating a documentation website using Markdoc and Next.js.
```
Comment (leerob) [inline examples/with-markdoc/README.md:3]: Copy you copy the same README format as the other examples? 🙏

**Comment by mfix-stripe** [comment]
I think this should be good now @leerob! Would you rather I name the folder `markdoc/` or `with-markdoc/`?

**Review by leerob — DISMISSED** [review]
Thank you!

**Review by leerob — COMMENTED** [review]
I think you'll need to add a `.gitignore` that matches the other examples.

**Comment by balazsorban44** [comment]
Heads up, there was another example being added in this PR: https://github.com/vercel/next.js/pull/36867

**Comment by mfix-stripe** [comment]
> Heads up, there was another example being added in this PR: #36850
> 
> I think we should only merge one of them, and preferably use TypeScript either way. 👍

Let's use this PR. We will get this updated to TypeScript. (Cc @hideokamoto)

**Comment by hideokamoto-stripe** [comment]
@mfix-stripe 
I've made a new PR to your branch.
https://github.com/mfix-stripe/next.js/pull/1

Please review and merge it. 👍

**Review by leerob — DISMISSED** [review]

**Comment by mfix-stripe** [comment]
@leerob had to fix Prettier, sorry!

**Review by ijjk — DISMISSED** [review]

**Review by ijjk — APPROVED** [review]

**Pull request merged**